Led.cpp

```diff
@@ -1,33 +1,51 @@
 #include "Led.h"
 
-Led::Led() : pin(pin) {
+Led::Led() : Pin(Pin) {
 }
 
-Led::Led(unsigned pin) : pin(pin) {
+Led::Led(unsigned pin) : Pin(pin) {
     //	pinMode(pin, OUTPUT);
 }
 
 void Led::setBrightness(unsigned percent) {
-    //	analogWrite(led, precent);
+    //  percent = map(percent, 0, 100, 0, 255);
+    //	analogWrite(led, percent);
     brightness = percent;
 }
 
 void Led::turnOn() {
-    //	digitalWrite(pin, HIGH);
+    //  analogWrite(led, brightness);
+    flag = 1;
 }
 
 void Led::turnOff() {
     //	digitalWrite(pin, LOW);
+    flag = 0;
 }
 
 void Led::blink(unsigned n, unsigned delay1, unsigned delay2) {
-    for (unsigned i = 0; i < n; i++) {
-        turnOn();
-        //		delay(delay1);
-        turnOff();
-        if (i == n) {
-            break;
+
+    if (flag == 1) {
+
+        for (unsigned i = 0; i <= n; i++) {
```

Comment:
if n == 1 then this loop executes twice which is not what we want. 
and n == 2 will blink 3 times and so on.